src/map/ai/helpers/pathfind.cpp

```diff
@@ -588,6 +590,42 @@ float CPathFind::GetRealSpeed()
 
             realSpeed *= 1.0f + speedMod / 100.0f;
         }
+
+        if (settings::get<bool>("map.USE_MOB_SPEED_BOOST_MULTIPLIER"))
+        {
+            if (auto* mobEntity = dynamic_cast<CMobEntity*>(m_POwner))
+            {
+                bool mobIsPlayerPet = mobEntity->PMaster && mobEntity->PMaster->objtype == TYPE_PC;
+
+                auto* battleTarget         = mobEntity->GetBattleTarget();
+                bool  targetWouldBeDrawnin = battleTarget && (distance(mobEntity->loc.p, battleTarget->loc.p) - battleTarget->m_ModelRadius) > mobEntity->GetMeleeRange();
+                bool  isEngaged            = mobEntity->animation == ANIMATION_ATTACK;
+
+                if (isEngaged && !mobIsPlayerPet && targetWouldBeDrawnin)
+                {
+                    // use default multiplier for most mobs
+                    float multiplier = settings::get<float>("map.DEFAULT_MOB_SPEED_BOOST_MULTIPLIER");
+
+                    // unless the mob has a custom multiplier
+                    if (mobEntity->getMobMod(MOBMOD_SPEED_BOOST_MULT) > 0)
+                    {
+                        multiplier = mobEntity->getMobMod(MOBMOD_SPEED_BOOST_MULT);
+                    }
+
+                    // clamp the multiplier between 1 and 5 to prevent underflows and overflows
+                    multiplier = std::clamp<float>(multiplier, 1.0f, 5.0f);
+
+                    // if some weight penalty (like gravity) then cut the boost multiplier
+                    // (for mobs with default boost of 2.5 then boost becomes 1.2)
+                    if (mobEntity->getMod(Mod::MOVE_SPEED_WEIGHT_PENALTY) > 0)
+                    {
+                        multiplier *= 0.48;
+                    }
+
+                    realSpeed *= multiplier;
```

Comment:
What if we did the multiplier as a whole number int modifier as a percentage? Default setting as 250, then in the weight penalty block you can just divide by 2 to 125, then at the end here you can do:
```cpp
multiplier /= 10;
realSpeed *= multiplier / 10.f;
```
This would handle both of the issues I mentioned above and the 2.5 to 1.2 adjustment relatively cleanly (personally not a fan of the 0.48 adjustment).

Reply:
This should now be fixed. Thanks for the comments!

Reply:
I think you want to do this last part as 2 divisions by 10 instead of 1 by 100. That way the .5 will get dropped. As it is now I think you'll get 125% instead of the intended 120%.

Reply:
> I think you want to do this last part as 2 divisions by 10 instead of 1 by 100. That way the .5 will get dropped. As it is now I think you'll get 125% instead of the intended 120%.
> 
> Otherwise everything looks good! 👍

My concern is that then you have unclear logic for the reader. I would rather just do the 0.48 multiplication if accuracy is more important, as that is clearer for the reader.

Reply:
If I may give my opinion, no matter * 0.48 or 2 steps of flooring /10, a comment is needed stating whats going on and why

Reply:
> If I may give my opinion, no matter * 0.48 or 2 steps of flooring /10, a comment is needed stating whats going on and why

Yeah for sure